File: packages/transform/src/features/protobuf/ProtobufIsDecodeTransformer.ts
import { ProtobufIsDecodeProgrammer } from "@typia/core";

import { ITransformProps } from "../../ITransformProps";
import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufIsDecodeTransformer {
  export const transform = (props: ITransformProps) =>
    GenericTransformer.scalar({
      ...props,
      method: "protobuf.isDecode",
      write: ProtobufIsDecodeProgrammer.write,
    });
}

File: packages/transform/src/features/protobuf/ProtobufIsDecodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufIsDecodeTransformer = protobufIsDecodeTransformerNamespace{}

type protobufIsDecodeTransformerNamespace struct{}

func (protobufIsDecodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufIsDecodeProgrammer.Write(call.TypeText)
}
